### PR TITLE
Add tests for shortcuts.dart API examples.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR contributes to https://github.com/flutter/flutter/issues/130459

### Description
- Adds tests for `examples/api/lib/widgets/shortcuts/shortcuts.0.dart`
- Adds tests for `examples/api/lib/widgets/shortcuts/shortcuts.1.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

**Review by bleroux — APPROVED** [review]
LGTM! Thanks for the contribution.

File: examples/api/test/widgets/shortcuts/shortcuts.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_api_samples/widgets/shortcuts/shortcuts.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('ShortcutsExampleApp', () {
```
Comment (TahaTesser) [inline examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:11]: Single` group` not necessary.

File: examples/api/test/widgets/shortcuts/shortcuts.1_test.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_api_samples/widgets/shortcuts/shortcuts.1.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('ShortcutsExampleApp', () {
```
Comment (TahaTesser) [inline examples/api/test/widgets/shortcuts/shortcuts.1_test.dart:11]: Single` group` not necessary.

File: examples/api/test/widgets/shortcuts/shortcuts.1_test.dart [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      expect(
+        find.text('Subtract from the counter by pressing the down arrow key'),
+        findsOneWidget,
+      );
+      expect(find.text('count: 0'), findsOneWidget);
+    });
+
+    testWidgets(
+      'updates counter on arrowUp and arrowDown press',
+      (WidgetTester tester) async {
```
Comment (TahaTesser) [inline examples/api/test/widgets/shortcuts/shortcuts.1_test.dart:31]:
```suggestion
    testWidgets('updates counter on arrowUp and arrowDown press', (WidgetTester tester) async {
```

File: examples/api/test/widgets/shortcuts/shortcuts.0_test.dart [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      expect(
+        find.text('Subtract from the counter by pressing the down arrow key'),
+        findsOneWidget,
+      );
+      expect(find.text('count: 0'), findsOneWidget);
+    });
+
+    testWidgets(
+      'updates counter on arrowUp and arrowDown press',
+      (WidgetTester tester) async {
```
Comment (TahaTesser) [inline examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:31]:
```suggestion
    testWidgets('updates counter on arrowUp and arrowDown press', (WidgetTester tester) async {
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Comment by ksokolovskyi** [comment]
Hi @bleroux @TahaTesser thanks a lot for the review!

File: examples/api/test/widgets/shortcuts/shortcuts.0_test.dart (hunk 4 above)
Comment (ksokolovskyi) [reply examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:31]: The length of this line is greater than 80 chars, so formatter separates it into two parts.  @TahaTesser is there any suggestion to ignore formatter in such cases?

Comment (TahaTesser) [reply examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:31]: If the single `group`, this line would align closer to start and if it still cross 80 chars, there are only couple of tests here.

Comment (TahaTesser) [reply examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:31]: When adding group, the extra indent is needed  for this test title but we just have single group we can remove the group.

File: examples/api/test/widgets/shortcuts/shortcuts.0_test.dart (hunk 1 above)
Comment (ksokolovskyi) [reply examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:11]: I used a group here to get the following test output, while not copy-pasting `ShortcutsExampleApp` into each test name:
```
✓ ShortcutsExampleApp displays correct labels
✓ ShortcutsExampleApp updates counter on arrowUp and arrowDown press
```

While without group the output will be the following:
```
✓ displays correct labels
✓ updates counter on arrowUp and arrowDown press
```

Is this fine for you?

Comment (TahaTesser) [reply examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:11]: I can understand the benefits of this. However, existing tests have more descriptive title and do not repeat the example name as it is already indicated example test file naming scheme.

E.g. This describe the example bevhaior better without group.
https://github.com/flutter/flutter/blob/774ef8acf4ce4d8de739371e279c4f8296edccb4/examples/api/test/material/button_style_button/button_style_button.icon_alignment.0_test.dart#L5-L10

Comment (TahaTesser) [reply examples/api/test/widgets/shortcuts/shortcuts.0_test.dart]: Instead of example title, it describes the actual property used to update the icons

**Comment by ksokolovskyi** [comment]
@TahaTesser thanks for the replies and examples. I updated the tests according to your feedback. Could you please take a look again?

File: examples/api/test/widgets/shortcuts/shortcuts.0_test.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      findsOneWidget,
+    );
+    expect(
+      find.text('Subtract from the counter by pressing the down arrow key'),
+      findsOneWidget,
+    );
+    expect(find.text('count: 0'), findsOneWidget);
+  });
+
+  testWidgets('updates counter on arrowUp and arrowDown press', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:28]:
```suggestion
  testWidgets(' Up and down arrow press updates counter', (WidgetTester tester) async {
```

File: examples/api/test/widgets/shortcuts/shortcuts.0_test.dart [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_api_samples/widgets/shortcuts/shortcuts.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('displays correct labels', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:11]:
```suggestion
  testWidgets('Verify correct labels are displayed', (WidgetTester tester) async {
```

**Review by TahaTesser — APPROVED** [review]
LGTM! With updated titles

File: examples/api/test/widgets/shortcuts/shortcuts.0_test.dart (hunk 5 above)
Comment (TahaTesser) [reply examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:28]: Similar title in other test

File: examples/api/test/widgets/shortcuts/shortcuts.0_test.dart (hunk 6 above)
Comment (TahaTesser) [reply examples/api/test/widgets/shortcuts/shortcuts.0_test.dart:11]: Similar title in other test

**Review by TahaTesser — APPROVED** [review]

**Comment by ksokolovskyi** [comment]
@bleroux @TahaTesser thanks for the review!

**Pull request merged**